### ISSUE TITLE
Isues when Disabling a JAVA function in Azure portal

### ISSUE THREAD

**Issue by getmckenzie** [Azure/Azure-Functions#605]
I am facing issues when I choose  to disable a funtion using JAVA on Azure portal. I raised this earlier on msdn forums - https://social.msdn.microsoft.com/Forums/azure/en-US/90e463c7-b006-4319-98af-9e9ed771d744/how-to-disable-a-function-in-azure-using-java?forum=AzureFunctions#90e463c7-b006-4319-98af-9e9ed771d744
 but I did not get any response.

Following up with MS team, I got the suggestion that the function needs to be restarted everytime it was  disabled. I followed the same and here are my observations with regards to the same.

1.     I was not able to replicate the issue anymore. The "restart" was also not required.
2.	Clicking on ‘Save” after updating the function.json  did not give an alert that the settings are saved. So, as a user, I am not sure if my changes are effective or not.
3.	The portal did not update “(disabled)” until I clicked the “refresh icon” explicitly or reloaded the page. This does not give the clarity to the user if the function has been disabled or not. 
4.	I could verify if the function is disabled only by  visiting the manage tab. I can see that the "disabled" is highlighted.
5.	After this step when I invoked the URL, I got a 404 as expected.
6.	Now I enabled the functionality again through the “Manage” tab:
![w1](https://user-images.githubusercontent.com/33299622/33279993-738c5ba6-d355-11e7-9fcd-36a80957b4fb.jpg)

 

You can see that the function is still showing as “disabled” even when it is actually “enabled. At this point, I am able to invoke the API as expected.

7.	I need to explicitly “refresh” the function  to see the “(disabled)” go away.

 
![w2](https://user-images.githubusercontent.com/33299622/33280039-a1187ce4-d355-11e7-9229-ab166a084666.jpg)

 
![w3](https://user-images.githubusercontent.com/33299622/33280055-b1a8435a-d355-11e7-9eb2-2b0c36486701.jpg)

Can you please check and let me know?

**Comment by christopheranderson** [comment]
@alrod and I confirmed this is a known issue in v2 and should be fixed in the next release. The current portal behavior expects "disable" to be done via app settings and v2 is still using function.json, so it's not being updated. You'll need to manually update your function.json in the interim. Long term, we'll be switching to app settings which persist between deployments for all portal behavior here.

Fixed in this commit: https://github.com/Azure/azure-webjobs-sdk/commit/fa20618c420b1fc7173afdaa4e8678c5aab1a5ae
